src/core/public/core_system.ts

```diff
@@ -233,7 +233,17 @@ export class CoreSystem {
         injectedMetadata,
         notifications,
         uiSettings,
-        workspaces,
+      });
+
+      // set default value for filtered nav links
+      const navLinksService = chrome.navLinks;
```

Comment:
From the code, the default value will be all the links that not filtered by featureGrup/feature flag/current workspace. I would suggest not adding the default value here, as it will trigger a unnecessary render of collapse_nav component.

Reply:
How about check the uiSetting of workspace here? I think the OSD should have its default nav render logic. workspaces is just a plugin of OSD. We can't move the default nav generate logic to workspace plugin. @ruanyl What do you think about that?

Reply:
To avoid unnecessary rendering, I have a new idea. We can find some way to call the `navLinks.getNavLinks$()` as default `navLinks$` property in `src/core/public/chrome/chrome_service.tsx`. We can update to pass `navLinks.getFilteredNavLinks$()` after the `filteredNavLinks` updated. So we don't need to subscribe `navLinks` update here.